spring-cloud-kubernetes-core/src/main/java/io/fabric8/spring/cloud/kubernetes/reload/EventBasedConfigurationChangeDetector.java

```diff
@@ -0,0 +1,100 @@
+package io.fabric8.spring.cloud.kubernetes.reload;
+
+import javax.annotation.PostConstruct;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.spring.cloud.kubernetes.config.ConfigMapPropertySource;
+import io.fabric8.spring.cloud.kubernetes.config.ConfigMapPropertySourceLocator;
+import io.fabric8.spring.cloud.kubernetes.config.SecretsPropertySource;
+import io.fabric8.spring.cloud.kubernetes.config.SecretsPropertySourceLocator;
+
+import org.springframework.core.env.AbstractEnvironment;
+import org.springframework.core.env.MapPropertySource;
+
+/**
+ * A change detector that subscribes to changes in secrets and configmaps and fire a reload when something changes.
+ */
+public class EventBasedConfigurationChangeDetector extends ConfigurationChangeDetector {
+
+    private ConfigMapPropertySourceLocator configMapPropertySourceLocator;
+
+    private SecretsPropertySourceLocator secretsPropertySourceLocator;
+
+    private EventWatcher eventWatcher;
```

Comment:
Do we really need EventWatcher?

It only manages 2 kind of watches, and its a one liner to create. For example, one could replace:

    eventWatcher.addWatch("config-maps-watch", k -> k.configMaps().watch(...))

with

    configMapWatch = kubernetesClient.configMaps().watch(...);

without loosing any of the provided functionality right?